### PR TITLE
[SPARK-10836] [SparkR] Added sort(x, decreasing, col, ... ) method to DataFrame

### PR DESCRIPTION
**Pull request by NarineK**
the sort function can be used as an alternative to arrange(... ).
As arguments it accepts x - dataframe, decreasing - TRUE/FALSE, a list of orderings for columns and the list of columns, represented as string names

for example:
sort(df, TRUE, "col1","col2","col3","col5") # for example, if we want to sort some of the columns in the same order

sort(df, decreasing=TRUE, "col1")
sort(df, decreasing=c(TRUE,FALSE), "col1","col2")

### CODE REVIEW

**Comment by shivaram** [comment]
Jenkins, ok to test

**Comment by shivaram** [comment]
cc @sun-rui

**Comment by shivaram** [comment]
@NarineK Thanks for the PR. Could you also add a unit test for this ?

**Comment by NarineK** [comment]
Fixed the R style issues + added test cases
Also I did some changes in the signature to make sure that sort function works both for R{base} and R{SparkR} packages

**Comment by shivaram** [comment]
Jenkins, retest this please

File: R/pkg/R/DataFrame.R [hunk 1]
```diff
@@ -1848,3 +1848,67 @@ setMethod("crosstab",
...
+#' @export
+#' @examples
+#'\dontrun{
+#' sc <- sparkR.init()
+#' sqlContext <- sparkRSQL.init(sc)
+#' path <- "path/to/file.json"
+#' df <- jsonFile(sqlContext, path)
+#' sort(df, TRUE, "col1","col2")
+#' sort(df, decreasing=TRUE, "col1")
+#' sort(df, decreasing=c(TRUE,FALSE), "col1","col2")
```
Comment (felixcheung) [inline R/pkg/R/DataFrame.R]: could you please an example with decreasing at the default value, ie.
`sort(df, "col1")`

Comment (NarineK) [inline R/pkg/R/DataFrame.R]: sure

File: R/pkg/R/DataFrame.R [hunk 2]
```diff
@@ -1848,3 +1848,67 @@ setMethod("crosstab",
...
+#' }
+setMethod("sort",
+          signature(x = "DataFrame"),
+          function(x, decreasing=FALSE, col, ...) {
```
Comment (felixcheung) [inline R/pkg/R/DataFrame.R]: also, could/should it support Column type like in `arrange(df, df$age)`, ie.
`sort(df, df$age)`

Comment (NarineK) [inline R/pkg/R/DataFrame.R]: Hi Felix,
we can of course, the only thing is, that I need to embrace df$age with a sorting order and right now I'm doing it with string concatenation. I can try to get the string columns name of  df$age. That would be one option. 

I'm not sure how I could concatenate the order to column object nicely. It might make the code look ugly. 
I could have
loop: 
   if (order == "asc") { asc(col)  } else {desc(col)}

but this is not nice.
Let me know what do you think .

Thanks,
Narine

Comment (NarineK) [inline R/pkg/R/DataFrame.R]: Or if the column is smth like : asc(df$col1) and we have another decreasing=False. This is going to be smth like: asc(asc(df$col1)) which doesn't make much sense.

Comment (felixcheung) [inline R/pkg/R/DataFrame.R]: I think we could have this similar to `arrange`? line 1310 of DataFrame.R

Comment (NarineK) [inline R/pkg/R/DataFrame.R]: Well, I added similar to arrange now.... 

for sort it is a little more difficult with the signature and input arguments. It should comply with R's signature and it shouldn't brake the existing one. I cannot explicitly specify the column types in the signature... This is braking the actual sort function. 
I did play a lot with all possible combinations I could have thought of, so that I don't brake the sort{base} since you also use it in other places in SparkR.

**Comment by sun-rui** [comment]
Originally, we implemented a "sortDF" function in SparkR DataFrame corresponds to the "sort" function of the Scala DataFrame. Later, "sortDF" was renamed to "arrange" to be more frendly with "dlypr" package.

My question here is that naming this helper function as "sort" is desirable? This will override the "sort" function in R base package. After this change, If an R user wants to use the "sort" in base package in SparkR, it has to use base::sort(). As sort() may be widely and commonly used function, is this preferrable?

Could we find a new name for it?

File: R/pkg/R/DataFrame.R [hunk 3]
```diff
@@ -1848,3 +1848,78 @@ setMethod("crosstab",
...
+              colDFOrderConcatList <- paste("list(", colDFOrderConcat, ")", collapse = "")
+
+              # builds columns of type Column, example: [[1]] Column Species ASC
+              #                                         [[2]] Column Petal_Length DESC
+              resCols <- eval(parse(t=colDFOrderConcatList))
+              jcols <- lapply(resCols, function(c) {
+                c@jc
+              })
+            } else {
+              jcols <- lapply(by, function(c) {
```
Comment (sun-rui) [inline R/pkg/R/DataFrame.R]: If we support Column parameters, then the "decreasing" parameter makes no sense. I prefer we don't support Column parameters (As the existing arrange() and orderBy() is flexible enough for this usage)

File: R/pkg/R/DataFrame.R [hunk 4]
```diff
@@ -1848,3 +1848,78 @@ setMethod("crosstab",
...
+              colDFOrderConcat <- paste(sortOrder, "(", colDFConcat, ")", collapse = ",")
+
+              # concatenates all ordered columns to a list
+              # example: "list(asc(x$Species), desc(x$Petal_Length))"
+              colDFOrderConcatList <- paste("list(", colDFOrderConcat, ")", collapse = "")
+
+              # builds columns of type Column, example: [[1]] Column Species ASC
+              #                                         [[2]] Column Petal_Length DESC
+              resCols <- eval(parse(t=colDFOrderConcatList))
+              jcols <- lapply(resCols, function(c) {
```
Comment (sun-rui) [inline R/pkg/R/DataFrame.R]: can call arrange() or orderBy() here.

File: R/pkg/R/DataFrame.R [hunk 5]
```diff
@@ -1848,3 +1848,78 @@ setMethod("crosstab",
...
+              # example: c("asc(x$Species)", "desc(x$Petal_Length)" )
+              colDFOrderConcat <- paste(sortOrder, "(", colDFConcat, ")", collapse = ",")
+
+              # concatenates all ordered columns to a list
+              # example: "list(asc(x$Species), desc(x$Petal_Length))"
+              colDFOrderConcatList <- paste("list(", colDFOrderConcat, ")", collapse = "")
+
+              # builds columns of type Column, example: [[1]] Column Species ASC
+              #                                         [[2]] Column Petal_Length DESC
+              resCols <- eval(parse(t=colDFOrderConcatList))
```
Comment (sun-rui) [inline R/pkg/R/DataFrame.R]: Instead of building expressions and use eval(), I would prefer something like:
  lapply(seq_len(length(decreasing)), function(i) {
       if (decreasing[[i]]) {
          desc(column(x, col[[i]]))
      } else {
           asc(column(x, col[ii]])
  }
}

Comment (NarineK) [inline R/pkg/R/DataFrame.R]: Hi sun-rui,

thank you for you comments. As I mentioned in my previous notes I'm not overriding any R functions. We can use 
sort(iris) and this will call R basic package. (But I guess some of R basic functions are now being overriden by SparkR e.g. summary ;) )
We need the sort function because our customers who worked longer with R cannot find the functions and signatures they used to work with. Also arrange doesn't allow me to specify the order in case the column names are strings.

another advantage of having decreasing argument is that I can have smth like this:

sort(df, TRUE, "col1","col2","col3", .... "col500") and this will apply to all columns and I don't have to add that desc or asc prefix to all 500 columns manually. Which is time consuming.

Thanks for you suggestion about lapply. I'll give a try with it.

I also think that sort with only character "column" names would be better. The users can use arrange if they work with Column Object, because also, as you said, it makes "decreasing" parameters useless.

Let me know what do you think, so that I can proceed. 

Thanks,
Narine

Comment (NarineK) [inline R/pkg/R/DataFrame.R]: One correction: I do override but not overwrite it
An example:
both sort(iris$Species) and sort(sparkDF, FALSE, "Species") work perfectly

but what I meant with summary is
summary(iris) is failing .... I have to do base:::summary(iris) to make it running  
and the problem is in standardGeneric ....

Comment (NarineK) [inline R/pkg/R/DataFrame.R]: I've made the changes with lapply ... . The code looks more elegant now. Thank you!

Also, I couldn't directly call arrange, because I have a list of columns and it doesn't work when I pass a list or an array to arrange(x,col,...). I had to somehow cast it as three dot and it wasn't feasible. If you know how to do it and can share it with me that would be great.

Let me know what do you think about my latest commit

Thanks,
Narine

File: R/pkg/R/DataFrame.R [hunk 6]
```diff
@@ -1848,3 +1848,62 @@ setMethod("crosstab",
...
+            # example: [[1]] Column Species ASC
+            #          [[2]] Column Petal_Length DESC
+            columns <- lapply(seq_len(length(decreasing)), function(i){
+              if (decreasing[[i]]){
+                desc(getColumn(x, by[[i]]))
+              }else{
+                asc(getColumn(x, by[[i]]))
+              }
+            })
+
```
Comment (sun-rui) [inline R/pkg/R/DataFrame.R]: The following logic can be replaced by:
do.call("arrange", x, columns)

Comment (NarineK) [inline R/pkg/R/DataFrame.R]: Thank you ~sun-rui !!!
I've added it! let me know if you have any more comments.
Thanks!

**Comment by felixcheung** [comment]
looks good!

**Comment by NarineK** [comment]
Thanks! Would you mind, please, assigning the corresponding jira to me and resolving it ?

**Comment by shivaram** [comment]
@sun-rui Could you take one more look ?

**Comment by sun-rui** [comment]
My concern is still the name. Maybe we can make it combine with arrange().

@NarineK, could you update the function description more accurate (param), also coding style issues in the dontrun {}.

**Comment by NarineK** [comment]
Hi sun-rui,
I'll do the correction with param. I'm not sure what do you mean with coding style issues for dontrun {}.
I did it exactly as arrange and added examples for sort. Is the issue with decreasing=TRUE. ??? 

What don't you exactly like in the function name ?
1. We are trying to support R function namings. Similarly, as merge and join are synonyms.
2. I'm not overwriting anything.
3. in arrange you allow the users to do the ordering embracement manually and in sort, we do automatically. What do you mean with combining ? do you want to add another decreasing argument in arrange  ? If yes then we might need to think about cases like asc(asc(col)) cols ???  

I personally think that it makes sense to keep it separately as is. But if you can exactly tell me what the problem is, I might be able to understand and work on it.

Thanks,
Narine

**Comment by felixcheung** [comment]
I think @sun-rui does have a point here. Generally in R, `sort` is for list & vector
https://stat.ethz.ch/R-manual/R-devel/library/base/html/sort.html

But for data.frame, `order` is used (in base) (or `arrange` in dplyr)
http://stackoverflow.com/questions/1296646/how-to-sort-a-dataframe-by-columns
http://www.cookbook-r.com/Manipulating_data/Sorting/
http://www.dummies.com/how-to/content/how-to-sort-data-frames-in-r.html

And IMO ordering with Column is actually more R-like
```
df[ order(df$weight, decreasing=TRUE), ]
df[ order(-df$weight), ]
```

**Comment by NarineK** [comment]
okay, I see. For columns we probably cannot add sort right now because the dataframe api doesn't have it. 
For my understanding arrange isn't doing sorting for string column names.
we can add that decreasing attribute to arrange like in your example, Felix. But then it will be useless for Column objects ???

**Comment by felixcheung** [comment]
I think `arrange` does work with both column name strings or Column:
```
setMethod("arrange",
          signature(x = "DataFrame", col = "characterOrColumn"),
```

Anyway, I am only pointing out `order` syntax + `[` slicing/subseting can possibly be more familiar to R users.

**Comment by NarineK** [comment]
order is for vectors:
http://www.inside-r.org/r-doc/base/order

**Comment by NarineK** [comment]
I cannot understand one thing why are we supporting string column types here:
setMethod("arrange",
          signature(x = "DataFrame", col = "characterOrColumn"),

if we cannot sort by them?

**Comment by NarineK** [comment]
okay, our customers had that sort for dataframes too but it seems that R doesn't have it for dataframe.
Now let's decide what do we want to do.

we can have that decreasing argument optional for string columns in orderBy or arrange. R doesn't have it but I can still see that it is logical to have.

Or we just remove string arguments.

**Comment by NarineK** [comment]
And we did support that sort function for dataframe because we couldn't have it for columns. I cannot add it to SparkR columns either ...

**Comment by olarayej** [comment]
@shivaram @felixcheung @sun-rui Thanks for your feedback!

I totally see your point with the naming (sort vs. arrange), but @NarineK's implementation has two advantages:

1) It supports string column names in both asc and desc order. In the current SparkR's implementation of arrange(), I couldn't do that:

arrange(df, desc("Species")) # fails

2) Boolean parameter 'decreasing' is useful. Right now, if you were to sort by 100 columns, all of them in descending order, you'll need to write 100 times, for each column: desc(data$col1), ...., desc(data$col100), whereas in @NarineK's implementation, it will suffice to specify decreasing=T.

I'm aware that plyr also takes functions asc/desc, probably because R was not designed with big data in mind. We've seen customer use cases with hundreds of thousands of columns.

Bottom line: I think these are two valid additions to Spark R, and since the code is ready and tested, it won't hurt. Let the user decide which function to use.

File: R/pkg/R/DataFrame.R [hunk 7]
```diff
@@ -1848,3 +1848,57 @@ setMethod("crosstab",
             sct <- callJMethod(statFunctions, "crosstab", col1, col2)
             collect(dataFrame(sct))
           })
+
+#' Sort
+#'
+#' Sort a DataFrame by the specified column(s).
+#'
+#' @param x A DataFrame to be sorted.
+#' @param by A character column indicating the field to sort on.
```
Comment (shivaram) [inline R/pkg/R/DataFrame.R]: These roxygen docs don't match the function below. Should this be `col` ?

File: R/pkg/R/DataFrame.R [hunk 8]
```diff
@@ -1848,3 +1848,57 @@ setMethod("crosstab",
...
+#' sort(df, decreasing=TRUE, "col1")
+#' sort(df, c(TRUE,FALSE), "col1","col2")
+#' }
+setMethod("sort",
+          signature(x = "DataFrame"),
+          function(x, decreasing=FALSE, col, ...) {
+
+            # all sorting columns
+            by <- c(col, ...)
+            if (length(decreasing) == 1){
```
Comment (shivaram) [inline R/pkg/R/DataFrame.R]: space after `)`

File: R/pkg/R/DataFrame.R [hunk 9]
```diff
@@ -1848,3 +1848,57 @@ setMethod("crosstab",
...
+              # in case only 1 boolean argument - decreasing value is specified,
+              # it will be used for all columns
+              decreasing <- rep(decreasing,length(by))
```
Comment (shivaram) [inline R/pkg/R/DataFrame.R]: space after `,`

File: R/pkg/R/DataFrame.R [hunk 10]
```diff
@@ -1848,3 +1848,57 @@ setMethod("crosstab",
...
+            } else if (length(decreasing) != length(by)){
```
Comment (shivaram) [inline R/pkg/R/DataFrame.R]: same as before, space between `)` and `{`

**Comment by shivaram** [comment]
Thanks @olarayej for summarizing the points. I am actually fine with having two verbs `arrange` and `sort` for the same functionality. But thing that I'd like to avoid is having subtly different syntaxes / capabilities in each of them. So in this case do you think there is a version of `arrange` we can have that will support string column names and a boolean list as well ?

**Comment by NarineK** [comment]
Hi guys,

I have been playing with different options of implementing this. 

1. The first thing I tried is to override arrange twice. Smth like this:
arrange(x, col="Column", ....)
arrange(x, col="character", ..., decreasing=FALSE) 
and this is even possible but there are some issues with generating documentation and finding the corresponding arrange.

2. As next I tried to have the following signature:
arrange(x, col="characterOrColumn", ..., decreasing="logical") 
and this is working fine. I had to add additional checks for decreasing since it is the last parameter. We need to make sure that it is being picked up correctly.

3. Since orderBy calls arrange we need some modifications there too.

4. and the last thing: I think that having deceasing at the end is not too bad. R has for base:order a similar signature:
order(..., na.last = TRUE, decreasing = FALSE)

@shivaram @olarayej @sun-rui  @felixcheung let me know what do you think.

**Comment by sun-rui** [comment]
@NarineK, I like your first option:
arrange(x, col="Column", ....)
arrange(x, col="character", ..., decreasing=FALSE) 

I don't understand your issues "some issues with generating documentation and finding the corresponding arrange."

Does the following not work?
setGeneric("arrange", function(x, col, ...) { standardGeneric("arrange") })

setMethod("arrange",
         signature(x = "DataFrame", col = "Column"),
        function(x, col, ...) {
            }

setMethod("arrange",
         signature(x = "DataFrame", col = "character"),
        function(x, col, ..., decreasing=FALSE) {
        }

**Comment by NarineK** [comment]
@sun-rui  the problem is in the following:

when we generate the documentation we also should be able to find it through help.
For example when we do '?arrange' this shows the arrange entries. E.g.
arrange in plyr
arrange in SparkR

Since my both arranges are in the same package - SparkR it is showing only one. Also R requires that I have different name and title for them. If I use the same name, the function documentations are getting over-written.
I am not sure how we could call out both arranges in SparkR under 1 entry or maybe under 2 separate ... 

e.g. the following will work
?arrange1
?arrange2

**Comment by sun-rui** [comment]
Could you take the select function of DataFrame as example? For an S4 generic function, we can have multiple S4 methods for it, and the documentation for them can be combined.

File: R/pkg/R/DataFrame.R [hunk 11]
```diff
@@ -1304,24 +1306,62 @@ setClassUnion("characterOrColumn", c("character", "Column"))
...
 #' @rdname arrange
+#' @export
+setMethod("arrange",
+          signature(x = "DataFrame", col="character"),
+          function(x, col, ..., decreasing=FALSE) {
+
+            # all sorting columns
+            by <- list(col, ...)
+
+            # extracting the last element and uses it as decreasing if it is boolean
```
Comment (sun-rui) [inline R/pkg/R/DataFrame.R]: remove this block of code. Sorting direction must be specified by "decreasing = ". No need to check if last element in "..." is decreasing or not.

File: R/pkg/R/DataFrame.R [hunk 12]
```diff
@@ -1304,24 +1306,62 @@ setClassUnion("characterOrColumn", c("character", "Column"))
 #' path <- "path/to/file.json"
 #' df <- jsonFile(sqlContext, path)
 #' arrange(df, df$col1)
-#' arrange(df, "col1")
 #' arrange(df, asc(df$col1), desc(abs(df$col2)))
+#' arrange(df, "col1")
+#' arrange(df, "col2", FALSE)
```
Comment (sun-rui) [inline R/pkg/R/DataFrame.R]: remove this line

File: R/pkg/R/DataFrame.R [hunk 13]
```diff
@@ -1304,24 +1306,62 @@ setClassUnion("characterOrColumn", c("character", "Column"))
...
+#' arrange(df, "col1", decreasing=TRUE)
```
Comment (sun-rui) [inline R/pkg/R/DataFrame.R]: coding style. decreasing = TRUE

File: R/pkg/R/DataFrame.R [hunk 14]
```diff
@@ -1304,24 +1306,62 @@ setClassUnion("characterOrColumn", c("character", "Column"))
...
+#' arrange(df, "col1", "col2", c(TRUE, FALSE))
```
Comment (sun-rui) [inline R/pkg/R/DataFrame.R]: decreasing = c(TRUE, FALSE)

File: R/pkg/R/DataFrame.R [hunk 15]
```diff
@@ -1304,24 +1306,62 @@ setClassUnion("characterOrColumn", c("character", "Column"))
...
 #' }
 setMethod("arrange",
-          signature(x = "DataFrame", col = "characterOrColumn"),
+          signature(x = "DataFrame", col="Column"),
```
Comment (sun-rui) [inline R/pkg/R/DataFrame.R]: coding style: col = "Column"

File: R/pkg/R/DataFrame.R [hunk 16]
```diff
@@ -1304,24 +1306,62 @@ setClassUnion("characterOrColumn", c("character", "Column"))
...
           function(x, col, ...) {
-            if (class(col) == "character") {
-              sdf <- callJMethod(x@sdf, "sort", col, list(...))
-            } else if (class(col) == "Column") {
               jcols <- lapply(list(col, ...), function(c) {
```
Comment (sun-rui) [inline R/pkg/R/DataFrame.R]: adjust indent.

File: R/pkg/R/DataFrame.R [hunk 17]
```diff
@@ -1304,24 +1306,62 @@ setClassUnion("characterOrColumn", c("character", "Column"))
...
-            }
+
+            sdf <- callJMethod(x@sdf, "sort", jcols)
             dataFrame(sdf)
           })
 
 #' @rdname arrange
+#' @export
+setMethod("arrange",
+          signature(x = "DataFrame", col="character"),
```
Comment (sun-rui) [inline R/pkg/R/DataFrame.R]: coding style: col = "character"

File: R/pkg/R/DataFrame.R [hunk 18]
```diff
@@ -1304,24 +1306,62 @@ setClassUnion("characterOrColumn", c("character", "Column"))
...
+            lastElement <- tail(by,1)[[1]]
+            if (is.logical(lastElement)) {
+              length(by) <- length(by) - 1
+              decreasing <- lastElement
+            }
+
+            if (length(decreasing) == 1) {
+              # in case only 1 boolean argument - decreasing value is specified,
+              # it will be used for all columns
+              decreasing <- rep(decreasing,length(by))
```
Comment (sun-rui) [inline R/pkg/R/DataFrame.R]: coding style: decreasing, length(by)

File: R/pkg/R/generics.R [hunk 19]
```diff
@@ -393,7 +393,7 @@ setGeneric("agg", function (x, ...) { standardGeneric("agg") })
 
 #' @rdname arrange
 #' @export
-setGeneric("arrange", function(x, col, ...) { standardGeneric("arrange") })
+setGeneric("arrange", function(x, col, ...) {standardGeneric("arrange") })
```
Comment (sun-rui) [inline R/pkg/R/generics.R]: coding style: { standardGeneric("arrange") })

File: R/pkg/inst/tests/test_sparkSQL.R [hunk 20]
```diff
@@ -992,6 +992,15 @@ test_that("arrange() and orderBy() on a DataFrame", {
...
   expect_equal(first(sorted4)$name, "Michael")
   expect_equal(collect(sorted4)[3,"name"], "Andy")
+
+  sorted5 <- arrange(df, "age", "name", TRUE)
+  expect_equal(collect(sorted5)[1,2], "Andy")
+
+  sorted6 <- arrange(df, "age","name", c(T, F))
+  expect_equal(collect(sorted6)[1,2], "Andy")
+
+  sorted7 <- arrange(df, "name", decreasing=FALSE)
```
Comment (sun-rui) [inline R/pkg/inst/tests/test_sparkSQL.R]: coding style: decreasing = FALSE

File: R/pkg/inst/tests/test_sparkSQL.R [hunk 21]
```diff
@@ -992,6 +992,15 @@ test_that("arrange() and orderBy() on a DataFrame", {
   sorted4 <- orderBy(df, desc(df$name))
   expect_equal(first(sorted4)$name, "Michael")
   expect_equal(collect(sorted4)[3,"name"], "Andy")
+
+  sorted5 <- arrange(df, "age", "name", TRUE)
+  expect_equal(collect(sorted5)[1,2], "Andy")
+
+  sorted6 <- arrange(df, "age","name", c(T, F))
```
Comment (sun-rui) [inline R/pkg/inst/tests/test_sparkSQL.R]: df, "age","name", decreasing = c(TRUE, FALSE)

**Comment by NarineK** [comment]
@sun-rui Thanks!
Done the changes, please check it out.

**Comment by NarineK** [comment]
also, I hope you won't mind if I change col type in orderBy from characterOrColumn to only Column.
This will make more sense....

**Comment by sun-rui** [comment]
@NarineK, thanks for your follow up. Now basically looks good. I'd like to wait to see other's comments.

As for orderBy, maybe we can keep it as is?

File: R/pkg/R/DataFrame.R [hunk 22]
```diff
@@ -1290,8 +1290,10 @@ setClassUnion("characterOrColumn", c("character", "Column"))
 #' Sort a DataFrame by the specified column(s).
 #'
 #' @param x A DataFrame to be sorted.
-#' @param col Either a Column object or character vector indicating the field to sort on
+#' @param col A character or Column object vector indicating the fields to sort on
 #' @param ... Additional sorting fields
+#' @param decreasing A logical argument indicating sorting order for columns which are specified
+#'                   with their string names
```
Comment (felixcheung) [inline R/pkg/R/DataFrame.R]: should we say something like `decreasing` is 'when a character vector is specified for col'?

**Comment by felixcheung** [comment]
looks good

**Comment by NarineK** [comment]
Hi @felixcheung ,

I try to say it  at param description:

 @param decreasing A logical argument indicating sorting order for columns which are specified
                   with their string names

**Comment by felixcheung** [comment]
Hi Narine - it looks like the `decreasing` param is only supported with `col` is a character vector right? it might help to point that out if an user is trying pass that when `col` is a Column

**Comment by felixcheung** [comment]
This is in reference to the previous doc line which says:
```
#' @param col A character or Column object vector indicating the fields to sort on
```
so it might be less direct to say `with their string names`

**Comment by NarineK** [comment]
Hi Felix, 

so do you suggest to change the comment with smth like: 

#' @param decreasing A logical argument indicating sorting order for columns when a character vector is specified for col

is this fine ?

**Comment by NarineK** [comment]
and I can leave only:
@param col A character or Column object vector indicating the fields to sort on 

for col

**Comment by felixcheung** [comment]
yes, that sounds good, thanks @NarineK

**Comment by NarineK** [comment]
any more comments ?

**Comment by sun-rui** [comment]
LGTM

**Comment by shivaram** [comment]
Thanks @NarineK for the PR and @sun-rui for reviewing this.

LGTM and merging this
